### PR TITLE
[Tooling] Refactor aptos-debugger and add two new tools.

### PR DESCRIPTION
**Pull request by grao1991**
### Description
Add a tool to dump block from consensus/quorum_store DBs.
Add a tool to execute uncommitted blocks in consensus/quorum_store DBs, at a given version.

### CODE REVIEW

File: crates/aptos-debugger/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ rust-version = { workspace = true }
 
 [dependencies]
 anyhow = { workspace = true }
+aptos-consensus = { workspace = true }
```
Comment (ibalajiarun) [inline crates/aptos-debugger/Cargo.toml:17]: Such dependencies make our tools too heavy. I think we should consider move `aptos-consensus-types` to a separate crate. (May be in a separate PR but I don't know where we actually track these things).

Comment (grao1991) [reply crates/aptos-debugger/Cargo.toml:17]: aptos-consensus-types is already a separate crate..

There are some other things need to be moved to separate crates to avoid this dependency, e.g. consensus/quorum db.. I will leave it for future PR.

**Review by msmouse — APPROVED** [review]
1. would be nice to be able to run individual txns from consensus / qs DBs
2. would be nice if ConsensusDbTool holds Arc<ConsensusDB> etc so we don't need to expose the DBs to the admin service or the db tool.

**Comment by grao1991** [comment]
> 1. would be nice to be able to run individual txns from consensus / qs DBs
> 2. would be nice if ConsensusDbTool holds Arc etc so we don't need to expose the DBs to the admin service or the db tool.

Agree. Will consider in a separate PR.

File: aptos-move/aptos-debugger/src/execute_pending_block.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            AptosDebugger::rest_client(Client::new(Url::parse(&rest_endpoint)?))?
+        } else if let Some(db_path) = self.opts.target.db_path {
+            AptosDebugger::db(db_path)?
+        } else {
+            unreachable!("Must provide one target.");
+        };
+
+        let user_txns = if let Some(block_rest_endpoint) = self.block_rest_endpoint {
+            info!(
+                "Getting block {:?} from {block_rest_endpoint:?}.",
```
Comment (zekun000) [inline aptos-move/aptos-debugger/src/execute_pending_block.rs:54]: what is this {block_reset_endpoint:?}?

File: consensus/src/util/db_tool.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                    for proof in &proof_with_data.proofs {
+                        let digest = proof.digest();
+                        if let Some(batch) = all_batches.get(digest) {
+                            if let Some(txns) = batch.payload() {
+                                block_txns.extend(txns);
+                            } else {
+                                bail!("Payload is not found for batch ({digest}).");
+                            }
+                        } else {
+                            bail!("Batch ({digest}) is not found.");
```
Comment (zekun000) [inline consensus/src/util/db_tool.rs:82]: cc @bchocho we discussed to persist remote fetched batches

File: consensus/src/util/db_tool.rs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                        }
+                    }
+                },
+            }
+            Ok(block_txns)
+        },
+        None => {
+            bail!("No payload in the block.")
```
Comment (zekun000) [inline consensus/src/util/db_tool.rs:90]: this should just return vec![] instead of bail, it can be NIL block

**Review by zekun000 — APPROVED** [review]

File: aptos-move/aptos-debugger/src/execute_pending_block.rs (hunk 2 above)
Comment (grao1991) [reply aptos-move/aptos-debugger/src/execute_pending_block.rs:54]: The endpoint in AdminService to get txns in consensus_db/quorumstore_db by block id.

File: consensus/src/util/db_tool.rs (hunk 3 above)
Comment (bchocho) [reply consensus/src/util/db_tool.rs:82]: Yup, will work on this. But to be clear, there are still cases (especially disasters) when the remote batch could be missing -- e.g., all the validators with the remote batch are currently offline.

File: consensus/src/util/db_tool.rs (hunk 4 above)
Comment (grao1991) [reply consensus/src/util/db_tool.rs:90]: Done
